image_rotate/CMakeLists.txt

```diff
@@ -1,32 +1,64 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 project(image_rotate)
 
-find_package(catkin REQUIRED COMPONENTS cmake_modules cv_bridge dynamic_reconfigure image_transport nodelet roscpp tf2 tf2_geometry_msgs)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-# generate the dynamic_reconfigure config file
-generate_dynamic_reconfigure_options(cfg/ImageRotate.cfg)
-
-catkin_package()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
+find_package(ament_cmake REQUIRED)
+find_package(cv_bridge REQUIRED)
+find_package(class_loader REQUIRED)
+find_package(image_transport REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_ros REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
 find_package(OpenCV REQUIRED core imgproc)
 
 # add the executable
-include_directories(${catkin_INCLUDE_DIRS} ${OpenCV_INCLUDE_DIRS})
+include_directories(include)
 
 add_library(${PROJECT_NAME} SHARED src/nodelet/image_rotate_nodelet.cpp)
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${OpenCV_LIBRARIES})
-add_dependencies(${PROJECT_NAME} ${PROJECT_NAME}_gencfg)
+target_link_libraries(${PROJECT_NAME} ${ament_LIBRARIES} ${OpenCV_LIBRARIES})
+ament_target_dependencies(${PROJECT_NAME}
+  "image_transport"
+  "cv_bridge"
+  "class_loader"
+  "rclcpp"
+  "tf2_ros"
+  "tf2_geometry_msgs"
+)
+rclcpp_register_node_plugins(${PROJECT_NAME} "${PROJECT_NAME}::ImageRotateNode")
+set(node_plugins "${node_plugins}${PROJECT_NAME}::ImageRotateNode;$<TARGET_FILE:ImageRotateNode>\n")
+
 install(TARGETS image_rotate
-        DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+        DESTINATION lib
 )
 
 add_executable(image_rotate_exe src/node/image_rotate.cpp)
```

Comment:
I know it was called this previously, but I'm not a big fan of the _exe designation, especially now that ROS2 supports Windows. This would be confusing for a Windows user (with .exe being the executable extension). I would prefer something like _node or _bin or pretty much anything else.

Reply:
Me too. 
fixed by https://github.com/ros-perception/image_pipeline/pull/385/commits/45bd9d037c3a83e888b9844eb2c74e1fe22cce74